url/src/comp.php

```diff
@@ -341,35 +337,88 @@ function replace_url($itemID, $oldURL, $newURL, $shortCode, $TalisGUID, $token){
-			// build the PATCH body
-			$body = build_patch_body($resource_id, $web_address_array, $newURL);
-			echo_message_to_screen(DEBUG, "replace_url patch request body: $body");
-			// if not a dry run - update
-			if ($shouldWritetoLive === "true") {
-				post_url($shortCode, $resource_id, $body, $TalisGUID, $token);
-			} else {
-				fwrite($myfile, "Test Run\r\n");
+			if ($web_address_array) {
+				$web_address_found = true;
+				$resource_updated = true;
+				// add a new web addresses to the existing ones
+				$new_web_address_array = check_web_addresses($oldURL, $newURL, $web_address_array, "replace");
+				// If the online resource url matches the old url then change
+				$new_online_resource = ($online_resource === $oldURL) ? $newURL : $online_resource;
+				// Nothing further to do if there are no changes to make
+				if($web_address_array === $new_web_address_array && $online_resource === $new_online_resource){
+					echo_message_to_screen(DEBUG, "Nothing to update with this resource</br>");
+					fwrite($myfile,"Nothing to update with this resource\r\n");
+					continue;
+				}
+				// build the PATCH body
+				$body = build_patch_body($resource_id, $new_web_address_array, $new_online_resource);
+				echo_message_to_screen(DEBUG, "replace_url patch request body: $body");
+				// if not a dry run - update
+				if ($shouldWritetoLive === "true") {
+					post_url($shortCode, $resource_id, $body, $TalisGUID, $token);
+				} else {
+					fwrite($myfile, "Test Run\r\n");
+				}
+				increment_counter('URLs replaced: ');
 			}
-			increment_counter('URLs replaced: ');
-		} else {
+		}
+		// We only want to flag these warnings if they do not happen for any resource in this item
+		if ($web_address_found === false){
+			echo_message_to_screen(INFO, "Web Address Array is empty\t");
+			fwrite($myfile,"Web Address Array is empty,");
+		}
+		if ($online_resource_found === false){
+			echo_message_to_screen(INFO, "Online Resource is not set\t");
+			fwrite($myfile,"Online Resource is not set,");
+		}
+		if ($resource_updated === false){
 			fwrite($myfile, "Resource URL Not Updated - Dry Run\r\n");
 		}
 	}
 
 }
 
+/**
+ * Extract all resources from the item data returned from the API
+ * There may be 0 or more resources found
+ *
+ * @param array $itemData
+ * @return array
+ */
+function get_resources_from_item(stdClass $itemData){
+	$resources = [];
+	if (!isset($itemData->included)){
+		return $resources;
+	}
+	foreach ($itemData->included as $v){
+		if(!isset($v->type)){
+			continue;
+		}
+		if($v->type == "resources"){
+			$resources[] = $v;
+		}
+	}
+	return $resources;
+}
+
 function get_item($shortCode, $itemID, $TalisGUID, $token) {
-	$item_lookup = "https://rl.talis.com/3/" . $shortCode . "/draft_items/" . $itemID . "?include=resource";
+	$item_lookup = "https://rl.talis.com/3/" . $shortCode . "/draft_items/" . $itemID . "?include=resource,resource.part_of";
```

Comment:
This will also grab the part_of data.  It will appear alongside as another resource in the "includes" array.